src/arena.c

```diff
@@ -1259,13 +1259,16 @@ static size_t
 arena_prof_demote(tsdn_t *tsdn, edata_t *edata, const void *ptr) {
 	cassert(config_prof);
 	assert(ptr != NULL);
+	size_t usize = isalloc(tsdn, ptr);
+	size_t bumped_usize = sz_sa2u(usize, PROF_SAMPLE_ALIGNMENT);
```

Comment:
Can we do this?

    size_t bumped_usize = edata_size_get(edata);
    assert(bumped_usize == sz_sa2u(usize, PROF_SAMPLE_ALIGNMENT));

Reply:
When it fails, what are the sizes?

Reply:
I ran it locally with `--enable-cache-oblivious` and was able to reproduce the failure. Re-running after applying your suggested change does indeed fix it.